src/main/webapp/app/overview/course-statistics/course-statistics.component.ts

```diff
@@ -531,49 +533,27 @@ export class CourseStatisticsComponent implements OnInit, OnDestroy, AfterViewIn
         presentationScores[ExerciseType.TEXT] = textExercisePresentationScore;
         presentationScores[ExerciseType.FILE_UPLOAD] = fileUploadExercisePresentationScore;
         this.presentationScoresPerExercise = presentationScores;
-        this.overallPresentationScore = this.calculateTotalScoreForTheCourse(ScoreType.PRESENTATION_SCORE);
+        this.overallPresentationScore = this.retrieveScoreTypeForExerciseType(ExerciseTypeTOTAL.TOTAL, ScoreType.PRESENTATION_SCORE);
     }
 
     /**
-     * Calculates the total score for every exercise in the course satisfying the filter function
-     * @param filterFunction the filter the exercises have to satisfy
-     * @returns map containing score for every score type
-     * @private
-     */
-    private calculateScores(filterFunction: (courseExercise: Exercise) => boolean): Map<string, number> {
-        let courseExercises = this.courseExercises;
-        if (filterFunction) {
-            courseExercises = courseExercises.filter(filterFunction);
-        }
-        return this.courseCalculationService.calculateTotalScores(courseExercises, this.course!);
-    }
-
-    /**
-     * Calculates an arbitrary score type for an arbitrary exercise type
-     * @param exerciseType the exercise type for which the score should be calculated. Must be an element of {Programming, Modeling, Quiz, Text, File upload}
-     * @param scoreType the score type that should be calculated. Element of {Absolute score, Max points,Current relative score,Presentation score,Reachable points,Relative score}
+     * Retrieves an arbitrary score type for an arbitrary exercise type from the scores storage service. Scores are calculated in the server when fetching all courses.
+     * @param exerciseType the exercise type for which the score should be retrieved. Must be an element of {Programming, Modeling, Quiz, Text, File upload} or the total score for all exercises.
+     * @param scoreType the score type that should be retrieved from the store. Element of {'absoluteScore', 'maxPoints', 'currentRelativeScore', 'presentationScore', 'reachablePoints', 'relativeScore'}
      * @returns requested score value
```

Comment:
According to the description, you retrieve a type, but here you return a value. Please clarify this.